### DISCUSSION TITLE
Weird behavior when attempting to insert a URL to GITHUB_STEP_SUMMARY

### DISCUSSION THREAD

**Discussion by thehig** [community/community#42321] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a github action that generates a HTML report and uploads said report to an external server over ssh.

I'm trying to insert a link to the external server to the Github Action Summary

```
      - name: Add test result link to github summary
        run: echo '### [Open Test Report](https://mydomain.example.com)' >> $GITHUB_STEP_SUMMARY
```

However, when this action runs, references to `https://mydomain.example.com` are replaced with `https://***.example.com`. Why is this?

**Comment by airtower-luna** [comment, marked as answer]
Do you possibly have a secret with the value `mydomain`? If yes, it's probably the automatic secret masking.

**Reply by thehig** [reply]
> Do you possibly have a secret with the value `mydomain`? If yes, it's probably the automatic secret masking.

Nope

> Sneaky edit: I was wrong

**Reply by airtower-luna** [reply]
Anything *similar*? The `***` looks a lot like it's the masking mechanism, and I don't see what else would modify the output there, though I don't know exactly how it captures "possible" secrets. :thinking:

**Reply by thehig** [reply]
You're 100% right. I looked into it further and it turns out one of the secrets *does* contain the subdomain. Thanks you for the help

**Reply by J1033511** [reply]
I see

**Comment by LoemChanndany** [comment]
Yes

**Reply by thehig** [reply]
Yes what?

**Comment by tking16** [comment]
Is there a way to stop this behaviour? Because the close `)` bracket is being replaced by `***` so the hyperlink is being completely messed up. There's too many secrets in our organisation for go and find and root it out
